cvxpy/reductions/expr2smooth/canonicalizers/abs_canon.py

```diff
@@ -43,14 +45,20 @@ def smooth_abs_canon(expr, args):
     
     return t1, [y ** 2 == np.ones(shape), t1 == multiply(y, args[0])]
 
-def approx_abs_canon(expr, args):
+def smooth_approx_abs_canon(expr, args):
     # smooth approximation of abs via
     # \frac{1}{a}\cdot\log\left(e^{ax}+e^{-ax}\right)
     shape = expr.shape
     t1 = Variable(shape, bounds = [0, None])
-    a = 1.0
+    a = 5.0
     if args[0].value is not None:
         t1.value = np.abs(args[0].value)
     
-    expr = (1/a) * log(exp(a*args[0]) + exp(-a*args[0]))
-    return t1, [t1 == expr]
+    _exp_pos = exp(multiply(a, args[0]))
+    _exp_neg = exp(multiply(-a, args[0]))
+    exp_pos, constr_pos = exp_canon(_exp_pos, _exp_pos.args)
+    exp_neg, constr_neg = exp_canon(_exp_neg, _exp_neg.args)
+    _log_expr = log(exp_pos + exp_neg)
+    log_expr, constr_log = log_canon(_log_expr, _log_expr.args)
+
+    return t1, [a * t1 == log_expr] + constr_pos + constr_neg + constr_log
```

Comment:
I wonder if there is a way to avoid calling the canon files manually. Let me think about it.
How is this different than say using multiply in a constraint of div? 
Oh I see, it's probably because we want to ensure the arguments are single variables.